code/go/0chain.net/blobbercore/handler/worker.go

```diff
@@ -177,55 +182,68 @@ func startMoveColdDataToCloud(ctx context.Context) {
-	fs := filestore.GetFileStore()
-	allocation, err := fs.SetupAllocation(fileRef.AllocationID, true)
+	fileObjectPath, err := filestore.GetFileStore().GetPathForFile(fileRef.AllocationID, fileRef.ContentHash)
 	if err != nil {
-		Logger.Error("Unable to fetch allocation with error", zap.Any("allocationID", fileRef.AllocationID), zap.Error(err))
+		logging.Logger.Error("Error while getting path of file", zap.Error(err))
+	}
+	err = filestore.GetFileStore().MinioUpload(fileRef.ContentHash, fileObjectPath)
+	if err != nil {
+		logging.Logger.Error("Error uploading cold data to cloud", zap.Error(err), zap.Any("file_name", fileRef.Name), zap.Any("file_path", fileObjectPath))
 		return
 	}
 
-	dirPath, destFile := filestore.GetFilePathFromHash(fileRef.ContentHash)
-	fileObjectPath := filepath.Join(allocation.ObjectsPath, dirPath)
-	fileObjectPath = filepath.Join(fileObjectPath, destFile)
+	if fileRef.ThumbnailHash != "" {
+		thumbnailPath := ""
+		if err := filestore.GetFileStore().MinioUpload(fileRef.ThumbnailHash, thumbnailPath); err != nil {
+			logging.Logger.Error("Error uploading cold thumbnail data to cloud", zap.Error(err))
 
-	err = fs.UploadToCloud(fileRef.ContentHash, fileObjectPath)
-	if err != nil {
-		Logger.Error("Error uploading cold data to cloud", zap.Error(err), zap.Any("file_name", fileRef.Name), zap.Any("file_path", fileObjectPath))
-		return
+			logging.Logger.Info("Removing file from cloud")
+			if err := filestore.GetFileStore().MinioDelete(fileRef.ContentHash); err != nil {
+				logging.Logger.Debug("Got Error while remove file from cloud", zap.String("file", fileRef.ContentHash))
+			}
+			return
+		}
 	}
 
 	fileRef.OnCloud = true
 	ctx = datastore.GetStore().CreateTransaction(ctx)
 	db := datastore.GetStore().GetTransaction(ctx)
 	err = db.Save(fileRef).Error
 	if err != nil {
-		Logger.Error("Failed to update reference_object for on cloud true", zap.Error(err))
+		logging.Logger.Error("Failed to update reference_object for on cloud true", zap.Error(err))
 		db.Rollback()
 		ctx.Done()
 		return
 	}
 
 	db.Commit()
 	ctx.Done()
-	Logger.Info("Successfully uploaded file to cloud", zap.Any("file_name", fileRef.Name), zap.Any("allocation", fileRef.AllocationID))
+	logging.Logger.Info("Successfully uploaded file to cloud", zap.Any("file_name", fileRef.Name), zap.Any("allocation", fileRef.AllocationID))
+
+	newCtx := datastore.GetStore().CreateTransaction(context.Background())
+	db = datastore.GetStore().GetTransaction(newCtx)
+
+	var contentHashSharingRefsCount int64
+	condRef := reference.Ref{AllocationID: fileRef.AllocationID, ContentHash: fileRef.ContentHash}
+	if err := db.Model(&reference.Ref{}).Where(&condRef).Count(&contentHashSharingRefsCount).Error; err != nil {
+		logging.Logger.Error("Failed to get count", zap.Error(err))
+		return
+	}
 
-	if config.Configuration.ColdStorageDeleteLocalCopy {
+	if contentHashSharingRefsCount < 2 && config.Configuration.ColdStorageDeleteLocalCopy {
```

Comment:
Have put condition `<= 1` instead of `< 2`. I had done `< 2`, as it is single instruction than for `<=1 `.
Anyway readability should not be forgotten.